sklearn/neighbors/tests/test_lmnn.py

```diff
@@ -0,0 +1,677 @@
+    lmnn_warm.fit(X_train, y_train)
+    transformation_warm_plus_one = lmnn_warm.components_
+
+    lmnn_cold = LargeMarginNearestNeighbor(n_neighbors=3, warm_start=False,
+                                           max_iter=n_iter, random_state=0)
+    lmnn_cold.fit(X_train, y_train)
+    transformation_cold = lmnn_cold.components_
+    lmnn_cold.max_iter = 1
+    lmnn_cold.fit(X_train, y_train)
+    transformation_cold_plus_one = lmnn_cold.components_
+
+    diff_warm = np.sum(np.abs(transformation_warm_plus_one -
+                              transformation_warm))
+    diff_cold = np.sum(np.abs(transformation_cold_plus_one -
+                              transformation_cold))
+
+    assert diff_warm < 2.0, "Transformer changed significantly after one " \
+                            "iteration even though it was warm-started."
+
+    assert diff_cold > diff_warm, "Cold-started transformer changed less " \
+                                  "significantly than warm-started " \
+                                  "transformer after one iteration."
+
+
+def test_max_impostors():
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, max_impostors=1,
+                                      impostor_store='list')
+    lmnn.fit(iris_data, iris_target)
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, max_impostors=1,
+                                      impostor_store='sparse')
+    lmnn.fit(iris_data, iris_target)
+
+
+def test_neighbors_params():
+    from scipy.spatial.distance import hamming
+
+    params = {'algorithm': 'brute', 'metric': hamming}
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, neighbors_params=params)
+    lmnn.fit(iris_data, iris_target)
+    components_hamming = lmnn.components_
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3)
+    lmnn.fit(iris_data, iris_target)
+    components_euclidean = lmnn.components_
+
+    assert not np.allclose(components_hamming, components_euclidean)
+
+
+def test_impostor_store():
+    k = 3
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=k,
+                                      init='identity', impostor_store='list')
+    lmnn.fit(iris_data, iris_target)
+    components_list = lmnn.components_
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=k,
+                                      init='identity', impostor_store='sparse')
+    lmnn.fit(iris_data, iris_target)
+    components_sparse = lmnn.components_
+
+    assert_array_almost_equal(components_list, components_sparse,
+                              err_msg='Toggling `impostor_store` results in '
+                                      'a different solution.')
+
+
+def test_callback():
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, callback='my_cb')
+    assert_raise_message(ValueError,
+                         '`callback` is not callable.',
+                         lmnn.fit, iris_data, iris_target)
+
+    max_iter = 10
+
+    def my_cb(transformation, n_iter):
+        rem_iter = max_iter - n_iter
```

Comment:
You could also check that the transformation has the right shape here (see https://github.com/scikit-learn/scikit-learn/pull/10058#discussion_r260760785)

Reply:
Done.